prov/verbs/src/verbs_cq.c

```diff
@@ -479,9 +479,20 @@ static struct fi_ops vrb_cq_fi_ops = {
 static void vrb_cq_progress(struct util_cq *util_cq)
 {
 	struct vrb_cq *cq;
+	struct vrb_domain *domain;
+	struct ofi_epollfds_event event;
+	int nfds;
 
 	cq = container_of(util_cq, struct vrb_cq, util_cq);
 	vrb_flush_cq(cq);
+
+	domain = container_of(util_cq->domain, struct vrb_domain, util_domain);
+	if (!domain->epoll_fd)
+		return;
+
+	nfds = ofi_epoll_wait(domain->epoll_fd, &event, 1, 0);
+	if (nfds > 0)
+		vrb_domain_process_async_events(domain, nfds);
```

Comment:
This adds a kernel wait call into the cq progress path.  It is very unclear why this is needed for logging.

Reply:
we need to poll the async_fd to know that an event is ready to retrieve, otherwise `ibv_get_async_event` is synchronous and may block until an event becomes available.

Reply:
I still don't get it.  If the intent is to log when an async event is retrieved, why isn't that just done where the existing code reads an async event?  These changes actively poll for async events in different places than the previous code, so there's a lot more going on here than simply trying to log an event.

Reply:
Unless I'm severely mistaken, the existing code is polling for cq events, not async events, and only during cq_sread (talking specifically about polling an fd with poll, not considering ibv_poll_cq in the cq_read path).  I did not think adding async events to that path would be sufficient as it's only happening during cq_sread.  If you believe that would be sufficient it's simple enough to move the polling there.

Reply:
I don't understand the intent.  If we just want to log async events, why not simply add the log where the code reads them?  This change is introducing wait support and async events in places where the previous code would never have generated an async event.  Why are you trying to *add* async events?  Why is *logging* trying to change how or where the code waits for events and where it polls for them?  Enabling this logging is significantly changing the behavior beyond simply writing output.

Reply:
I'm not *adding* async events. I'm *adding* **retrieving and logging** of async events generated by the kernel.  These events are not captured in today's implementation. They are retrievable only with synchronous APIs that will block until events are available, and not done today. The alternative to synchronously getting the events is polling the async_fd to know there are events waiting, making the synchronous calls return immediately.

Understanding that this could impact performance and as it is primarily focused on aiding debugging, it's optional.  This poll will only happen when requested by the added environment variable.

Reply:
I can update the commit message to make the intent more clear and explain why we need to add a poll but I don't see a way to add support for this feature without adding a poll to the progress path as that is the path taken with fi_cq_read.